src/SourceGenerators/Directory.Build.targets

```diff
@@ -0,0 +1,47 @@
+<Project>
+
+  <!-- Extend Directory.Build.* from parent directory -->
+  <Import Project="..\$(MSBuildThisFile)" />
+
+  <!-- Common properties for all _packable_ source generation projects (all except CommonCode) -->
+  <PropertyGroup Condition="'$(IsPackable)' == 'true'">
+    <IsRoslynComponent>true</IsRoslynComponent>
+  </PropertyGroup>
+
+  <!-- Common code files are only visible in CommonCode project, to avoid cluttering the solution tree. -->
+  <PropertyGroup>
+    <CommonCodeVisible Condition="'$(CommonCodeVisible)' != 'true'">false</CommonCodeVisible>

```

Comment:
Why not just use linked files? Or directly a project reference? Not sure this is worth the added complexity... (always thinking about newcomers that might want to use this project as an example of authoring source generators or to contribute further)...

Reply:
As a newcomer to the project myself, I found it annoying to have common linked files interspersed with files specific to the project. Just my opinion, of course.

To mitigate the surprise factor as much as possible, I simplified and more extensively commented the `Directory.Build.targets` file.

Given the relative complexity of the solution, an onboarding document (`contributing.md` for example) may be a good place to explain this and other peculiarities.